exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -280,17 +280,17 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 		return err
 	}
 
-	// Initialize the patch helper.
-	patchHelper, err := patch.NewHelper(clusterResourceSetBinding, r.Client)
-	if err != nil {
-		return err
-	}
+	patch := client.MergeFromWithOptions(clusterResourceSetBinding.DeepCopy(), client.MergeFromWithOptimisticLock{})
+
+	var errList []error
 
 	defer func() {
 		// Always attempt to Patch the ClusterResourceSetBinding object after each reconciliation.
-		if err := patchHelper.Patch(ctx, clusterResourceSetBinding); err != nil {
-			log.Error(err, "Failed to patch config")
+		if patchErr := r.Client.Patch(ctx, clusterResourceSetBinding, patch); patchErr != nil {
+			log.Error(patchErr, "Failed to patch clusterResourceSetBinding")
```

Comment:
Let's add the Binding object to the log line
```suggestion
			log.Error(patchErr, "Failed to patch clusterResourceSetBinding", "clusterResourceSetBinding", klog.Kobj(clusterResourceSetBinding))
```